server/images.go

```diff
@@ -250,19 +250,21 @@ func GetModel(name string) (*Model, error) {
 		Template:  template.DefaultTemplate,
 	}
 
-	filename, err := GetBlobsPath(manifest.Config.Digest)
-	if err != nil {
-		return nil, err
-	}
+	if manifest.Config.Digest != "" {
```

Comment:
wouldn't be better to leave manifest.Config as a pointer and check nil? I don't see a benefit in changing to a value

Reply:
fwiw a nil manifest.Config should be considered abnormal, at least right now, and should only happen if the manifest was created manually

Reply:
The concern is that it is relatively easy to forget a nil check in the future given that with the current manifest format, this is likely to occur only with corruption. If we were able to have a single place where we checked it after deserialization then I would have kept it as a pointer but this requires checks in several places.

Changing to a value generally results in error messages if a check is missed rather than a crash. There are also fewer places that need checks when Config is stored as a value because some of them are OK with zero values (i.e. Sum()).

Reply:
I think value is okay, would just confirm that there are no new side effects elsewhere (e.g. where `.Digest` might be an empty string)

Reply:
I don't think that Digest should ever be an empty string in a non-error case besides the Config changes contemplated for the future. However, if it were to happen then the effects would essentially be the same as a missing Config, which is say we are already trying to handle this gracefully with this patch. And it is arguably more correct to not try to process a layer with an empty Digest anyways.

Reply:
That makes sense